WordPress/gutenberg
# Bootstrap an automation tool

We start to have a lot of manual tasks when performing plugin releases, npm releases, changelogs,... and all of these flows. I think we should invest some time automating these to avoid mistakes and I think it could be a huge gain in time ultimately.

To do so, I'd like us to change the "manual" required work from *performing the task* to *checking that everything went well and confirming* using an interactive CLI tool. This is obviously not an easy tool to build with all these flows but I'm confident it will bring more confidence and time gains.

I started looking at the plugin RC release process and bootstrapped a `release-plugin-rc` CLI tool. This flow is complex, this PR implements the following flow:

 - Checking the status of the branch and local changes.
 - Choosing the right version for the RC.
 - Creating the release branch.
 - Performing the bump (updated the files).
 - Committing the version bump changes. 
 - Building the plugin zip. 
 - Pushing the release branch and the git tag.
 - Creating a github release with the changelog and the zip.
 - Cherry-picking the version bump to master.

The process is covered entirely, we could push it a bit further and include slack notifications. 

This tool uses under the hood some npm utilities (commander, inquirer, chalk, simplegit), this adds a small burden when performing `npm install` on the project so I considered whether this should be a separate `package.json`/tool independent from the repository that you can bootstrap when needed. (This could be an easy follow-up if ever this grows too much).

Thoughts?

**Testing instructions**

The best way to try this locally without touching the official repo is to:

 1- Checkout this branch.
 2- Change the value of the `repoOwner` variable (set your username instead of WordPress).
 3- have a fork the Gutenberg repository.
 4- Run the command `bin/commander.js rc`.

At that point, your forked repo should be updated with the release.
